packages/keycloak-user-management/src/components/CreateEditUser/index.tsx

```diff
@@ -150,21 +122,19 @@ interface DispatchedProps {
 }
 
 // connect to store
-const mapStateToProps = (state: Partial<Store>, ownProps: PropsTypes): DispatchedProps => {
-  const userId = ownProps.match.params.userId;
-  const keycloakUsersSelector = makeKeycloakUsersSelector();
-  const keycloakUsers = keycloakUsersSelector(state, { id: [userId] });
-  const keycloakUser = keycloakUsers.length === 1 ? keycloakUsers[0] : null;
+const mapStateToProps = (state: Partial<Store>, ownProps: CreateEditPropTypes): DispatchedProps => {
+  const userId = ownProps.match.params[ROUTE_PARAM_USER_ID];
+  let keycloakUser = null;
+
+  if (userId) {
+    const keycloakUsersSelector = makeKeycloakUsersSelector();
+    const keycloakUsers = keycloakUsersSelector(state, { id: [userId] });
+    keycloakUser = keycloakUsers.length === 1 ? keycloakUsers[0] : null;
+  }
+
   const accessToken = getAccessToken(state) as string;
-  return { keycloakUser, accessToken };
-};
 
-/** map props to action creators */
-const mapDispatchToProps = {
-  fetchKeycloakUsersCreator: fetchKeycloakUsers,
+  return { keycloakUser, accessToken };
 };
 
-export const ConnectedCreateEditUsers = connect(
-  mapStateToProps,
-  mapDispatchToProps
-)(CreateEditUsers);
+export const ConnectedCreateEditUser = connect(mapStateToProps)(CreateEditUser);
```

Comment:
i have a question why dont we use redux hooks instead of doing it the old fashioned way 🤔💭 . what are the pros and cons for it

Reply:
@ashfahan I haven't used redux hooks before. What are the advantages over connect?

Reply:
the one thing i liked about it is the simplicity it have


> `connect` is a `High Order Component` whose job is to provide a way to connect Redux's store to your `components`. `useSelector` and `useDispatch` are the equivalent `hooks`. Just another technique to do the same thing.

> class Component extends React.Component{
    componentDidMount(){
        const { fetchApi, arg } = this.props
        fetchApi(arg)
    }
}
const mapDispatchToProps = dispatch =>({
    fetchApi : arg => dispatch(fetchApi(arg))
})
const mapStateToProps = state =>({
    arg : state.arg
})

> export default connect(mapStateToProps, mapDispatchToProps)(Component)
Now the equivalent using `hooks`

> const Component = () =>{
    const dispatch = useDispatch()
    const arg = useSelector(state => state.arg)

    useEffect(() =>{
       dispatch(fetchApi(arg))
    },[dispatch, arg])
}
> Both do exactly the same thing, it's only a different approach to connect `redux` to `components` internal `state`. Both consume Redux's context to expose `dispatch` and `state` inside a given `component`

source : https://stackoverflow.com/questions/58027300/what-is-the-main-difference-between-using-react-redux-hooks-and-react-redux-conn#:~:text=connect%20is%20a%20High%20Order%20Component

Reply:
@kelvin-muchiri @ashfahan since `connect` is still a valid way to do this, I do not think we need to refactor this now, I would say its an opt in pattern that we can start using on future prs or on future refactors, just my 2 cents. Though I eave it to you @kelvin-muchiri to choose which way you want to go.

Reply:
@ashfahan can you create an issue for the redux hooks pattern so that we can easily track this